packages/toolpad-app/src/toolpad/AppEditor/StructureExplorer/index.tsx

```diff
@@ -135,7 +135,12 @@ function RecursiveSubTree({
   }
   if (renderItem.length) {
     return (
-      <TreeItem nodeId={root.id} label={<Typography variant="body2">{root.name}</Typography>}>
+      <CustomTreeItem
+        nodeId={root.id}
+        node={root}
+        label={<Typography variant="body2">{root.name}</Typography>}
+        onHover={onHover}
```

Comment:
Forgot `onMouseLeave={onMouseLeave}` here? I feel like there's a lot of duplication in this component and the next developer may forget to pass a new property to every item here.

Reply:
It feels a bit like there's a lot of duplication in this component that leads to these sort of mistakes. Any way this could be improved?

Reply:
I'll look into that as well